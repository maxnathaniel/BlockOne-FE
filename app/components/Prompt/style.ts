import styled from '../../styles/styled-components';

export const PromptContainer = styled.div`
  padding: 5px 20px;
  width: 95%;
  background-color: #272F4D;
  margin: 20px auto;
  color: #66FCF1;
  min-height: 140px;
  max-height: 200px;

  @media all and (max-width: 800px) {
    width: 90%;
  }
`;

export const Header = styled.h3`
  margin-top: 10px !important;
  text-align: left;
`;

export const Body = styled.p`
  text-align: left;
  font-size: 1em;
`;

export const ButtonsContainer = styled.div`
  display: inline-block;
  margin: 5px 0;
  text-align: left;
`;