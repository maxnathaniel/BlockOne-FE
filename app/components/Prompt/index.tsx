import React, { useState, useEffect } from 'react';
import { Button } from 'semantic-ui-react';

import { Body, ButtonsContainer, Header, PromptContainer } from './style';

interface IPromptProps {
  header: string;
  message: string;
  onClickHandler: (e: any) => void;
}

const Prompt: React.FC<IPromptProps> = (props) => {
  const { onClickHandler } = props;

  return (
    <PromptContainer>
      <Header>
        {props.header}
      </Header>
      <Body>
          {props.message}
      </Body>
      <ButtonsContainer>
        <Button
          name="Create New Account"
          color="teal"
          content="Create New Account"
          size="medium"
          onClick={onClickHandler}
        />
      </ButtonsContainer>
    </PromptContainer>
  );
};

export default Prompt;