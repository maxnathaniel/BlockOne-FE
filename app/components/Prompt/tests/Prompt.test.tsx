import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Prompt from '../index';

const onClickHandlerFn = jest.fn();

describe('<Prompt />', () => {
  it('should render the various inner Elements', () => {
    const { container, getByText }
      = render(<Prompt header="Test Header" message="Welcome" onClickHandler={onClickHandlerFn} />);
    expect(getByText('Test Header')).toBeInTheDocument();
    expect(getByText('Welcome')).toBeInTheDocument();
    expect(getByText('Create New Account')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should invoke onClickHandlerFun on click', () => {
    const { container, getByText }
      = render(<Prompt header="Test Header" message="Welcome" onClickHandler={onClickHandlerFn} />);
    const btn = getByText('Create New Account');
    btn.click();
    expect(onClickHandlerFn).toBeCalled();
    expect(container).toMatchSnapshot();
  });
});