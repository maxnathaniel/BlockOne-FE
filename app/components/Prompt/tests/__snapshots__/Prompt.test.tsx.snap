// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Prompt /> should invoke onClickHandlerFun on click 1`] = `
<div>
  <div
    class="sc-AykKC ePKNZv"
  >
    <h3
      class="sc-AykKD cYEpnF"
    >
      Test Header
    </h3>
    <p
      class="sc-AykKE bkDhgi"
    >
      Welcome
    </p>
    <div
      class="sc-AykKF jvvefI"
    >
      <button
        class="ui teal medium button"
        name="Create New Account"
      >
        Create New Account
      </button>
    </div>
  </div>
</div>
`;

exports[`<Prompt /> should render the various inner Elements 1`] = `
<div>
  <div
    class="sc-AykKC ePKNZv"
  >
    <h3
      class="sc-AykKD cYEpnF"
    >
      Test Header
    </h3>
    <p
      class="sc-AykKE bkDhgi"
    >
      Welcome
    </p>
    <div
      class="sc-AykKF jvvefI"
    >
      <button
        class="ui teal medium button"
        name="Create New Account"
      >
        Create New Account
      </button>
    </div>
  </div>
</div>
`;
